src/shared/helpers.ts

```diff
@@ -49,10 +49,19 @@ export const getWalletInformation = async (
       Accept: 'application/json',
     },
   });
-  const json = await response.json();
+  if (!response.ok) {
+    if (response.status === 404) {
+      throw new Error('This wallet address does not exist.');
```

Comment:
Not throwing `new ErrorWithKey(key)` here as there's more consumers, and handling things there is out of scope of this PR.